### PR TITLE
TDD practice

### PR DESCRIPTION
**Pull request by ambrose-kibet**
## In this milestone we:
- [ ] Create a class called Solver  its  tests
- [ ] Create a method called factorial  and its  tests
- [ ] Create a method called fizzbuzz and its tests

### CODE REVIEW

**Review by amon-cofie — CHANGES_REQUESTED** [review]
Hi @ambrose-kibet and @Yemisirach , 👋 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter errors. Please fix these linter errors and resubmit for a review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -2,19 +2,50 @@
...
 
-    describe 'factorial method' do
-      it 'shoul retur 120 when a num of 5 is passed as an argument' do
+  describe 'factorial method' do
+    it 'shoul retur 120 when a num of 5 is passed as an argument' do
       expect(@solver.factorial(5)).to be(120)
     end
+    it 'shoul raise error when a num of -1 is passed as an argument' do
+      expect { @solver.factorial(-1) }.to raise_error(ArgumentError)
+    end
```
Comment (omarsalem7) [inline spec/solver_spec.rb:23]: - [ ] Kindly consider adding the case when the input is 0, the result should equal 1

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @ambrose-kibet, @Yemisirach 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🥳

Hi @ambrose-kibet, @Yemisirach 🙋‍♀️

Great work implementing the changes suggested by the reviewer 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ All tests are passing, perfect!
-  ✅ Great work using the right Gitflow 
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] I think it would be better to add Run tests section in README because it is related to this project. What do you think?😃

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**